src/self_driving/modeling/operating_unit_recorder.cpp

```diff
@@ -353,7 +353,7 @@ void OperatingUnitRecorder::AggregateFeatures(selfdriving::ExecutionOperatingUni
   std::vector<execution::translator_id_t> translator_ids;
   common::ManagedPointer<execution::compiler::OperatorTranslator> translator = current_translator_;
   translator_ids.emplace_back(translator->GetTranslatorId());
-  while (translator->IsCountersPassThrough()) {
+  while (translator->GetParentTranslator() != nullptr && translator->IsCountersPassThrough()) {
```

Comment:
@17zhangw I have no recollection of this code, do you know if the first condition would be masking a more serious issue?

Reply:
It _seems_ fine to me...

Reply:
> It seems fine? I’m a little skeptical of the plans being generated though, cause this seems to imply a series of top-level projections

-- @17zhangw